tests/ui/lint/noop-method-call.stderr

```diff
@@ -105,5 +105,57 @@ LL + #[derive(Clone)]
 LL | struct PlainType<T>(T);
    |
 
-warning: 7 warnings emitted
+warning: call to `.clone()` on a reference in this situation does nothing
+  --> $DIR/noop-method-call.rs:70:6
+   |
+LL |     v.clone();
+   |      ^^^^^^^^ help: remove this redundant call
+   |
+   = note: the type `non_clone_types::NotClone` does not implement `Clone`, so calling `clone` on `&non_clone_types::NotClone` copies the reference, which does not do anything and can be removed
+
+warning: call to `.clone()` on a reference in this situation does nothing
+  --> $DIR/noop-method-call.rs:75:6
+   |
+LL |     v.clone();
+   |      ^^^^^^^^
+   |
+   = note: the type `non_clone_types::ConditionalClone<PlainType<u32>>` does not implement `Clone`, so calling `clone` on `&non_clone_types::ConditionalClone<PlainType<u32>>` copies the reference, which does not do anything and can be removed
+help: remove this redundant call
+   |
+LL -     v.clone();
+LL +     v;
+   |
+help: if you meant to clone `non_clone_types::ConditionalClone<PlainType<u32>>`, implement `Clone` for it
+   |
+LL + #[derive(Clone)]
+LL | struct PlainType<T>(T);
+   |
+
+warning: call to `.clone()` on a reference in this situation does nothing
+  --> $DIR/noop-method-call.rs:80:6
+   |
+LL |     v.clone();
+   |      ^^^^^^^^ help: remove this redundant call
+   |
+   = note: the type `non_clone_types::ConditionalClone<non_clone_types::NotClone>` does not implement `Clone`, so calling `clone` on `&non_clone_types::ConditionalClone<non_clone_types::NotClone>` copies the reference, which does not do anything and can be removed
+
+warning: call to `.clone()` on a reference in this situation does nothing
+  --> $DIR/noop-method-call.rs:89:6
+   |
+LL |     v.clone();
+   |      ^^^^^^^^
+   |
+   = note: the type `non_clone_types::DifferentlyConditionalClone<PlainType<u8>>` does not implement `Clone`, so calling `clone` on `&non_clone_types::DifferentlyConditionalClone<PlainType<u8>>` copies the reference, which does not do anything and can be removed
+help: remove this redundant call
+   |
+LL -     v.clone();
+LL +     v;
+   |
+help: if you meant to clone `non_clone_types::DifferentlyConditionalClone<PlainType<u8>>`, implement `Clone` for it
```

Comment:
I'm not sure why this is being suggested, since the bound is `T: Default` instead of `T: Clone` here.

Reply:
It looks like the problem is that `cx.tcx.predicates_of(trait_impl_id)` is actually empty in this case.

Reply:
Same with this, you can modify
```
    .derive_suggestion = if you meant to clone `{$orig_ty}`, implement `Clone` for it
to be
```
    .derive_suggestion = if you meant to clone `{$non_clone_ty}`, implement `Clone` for it
```